### DISCUSSION TITLE
Slow corners

### DISCUSSION THREAD

**Discussion by Psylenceo** [supermerill/SuperSlicer#2437] [Q&A]
I started using Superslicer 2.3.57.7 on my original Makergear M2 that has been modified with a Duet3d Duet 3 motherboard a few months ago. I was able to configure everything to match my modifications and even went through the calibration wizard everything was fine. But, after closing superslicer then loading in some of my .3mf files from Fusion, when I tried printing straight lines would print at the correct speed. But, when the print gets to a corner, and this will always happen even on the brim, which is where I always noticed it, the speed drops to basically zero while the extruder is still going full speed creating massive blobs. I tried every slicer setting I could think of and nothing would fix this.  The ONLY way I could get prints to work was to load in a wizard print, delete the part then add my designed part then everything was fine.

Today, I decided to upgrade to 2.3.57.11, copied over the settings and I'm still getting the slow down on the corners, I tried loading the thermal tower and then removing the part but keeping the settings and still getting corners slowed down and blobing.

Would anyone have any idea what settings would affect this?

**Comment by neophyl** [comment]
That would suggest that its something in your settings that you are loading in.  If the calibration ones print fine then have you considered doing a comparison between them to see what exactly is different ?

Also as usual when dealing with anything like this, attaching a zipped up project file saved from SuSi goes a long way to make it possible to track down.  Without one its almost impossible to figure things out.

**Reply by Psylenceo** [reply]
I thought about attaching stuff but I wasn't sure which files would be the best. As for comparing a known good file with nonfunctional files, I was able to determine the setting that was causing the issue. It was due to the only custom start g-code option. Not sure why the slicing algorithm would get messed up because of it. Anyways, here's the g-code from a functional print:
```
M486 T5
; plater:{"center":[104.375000,125.000000,0.000000],"boundingbox_center":[104.375000,125.000000,0.662500],"boundingbox_size":[59.224998,100.949997,1.325000]}

M140 S70 ; set bed temperature
G10 P0 R223 ; sets the standby temperature
G10 P0 S223 ; set temperature
;TYPE:Custom
M98 P/macros/Print_Start_Macro
M107
G21 ; set units to millimeters
G90 ; use absolute coordinates
M82 ; use absolute distances for extrusion
G92 E0
; Filament gcode
G10 P0 S223 ; set temperature
M116 ; wait for temperature to be reached
M190 S70 ; set bed temperature and wait for it to be reached
M107
;LAYER_CHANGE
;Z:0.26
;HEIGHT:0.26
G1 Z0.26 F7800
```

Then from a non-functional g-code:
```
M486 T2
; plater:{"center":[100.000000,100.000000,0.000000],"boundingbox_center":[100.000000,100.000000,4.500000],"boundingbox_size":[22.500000,50.999999,9.000000]}

;TYPE:Custom
G28 ; home all axes
M98 P/macros/Print_Start_Macro
; Filament gcode
;LAYER_CHANGE
;Z:0.3
;HEIGHT:0.3
G1 E-1 F2400
```

And because I know someone will ask about the M98 start_macro I use:
```
M290 S{-move.axes[2].babystep}	;zero out babysteps
G90	;set absolute positioning
M82	;set extruder to absolute
M106 S0	;turn top fan off
M140 S70	;set bed to 70c
M190 S70	;wait for bed to reach 70c
if (global.completedPrints - global.OldPrintCount) >= 5
	set global.OldPrintCount = global.completedPrints	
	if abs(move.axes[2].babystep*100) != 0	;check if babysteps are still present
		M290 S{-move.axes[2].babystep}	;zero out babysteps
	set global.oldBedOffset = abs(abs(move.compensation.meshDeviation.deviation) + abs(move.compensation.meshDeviation.mean))/2	;calculate the old bed offset
	G90		;set to ABS mode
	M104 S100 T0		;heat extruder to 100c
	M109 S100 T0   		;wait for extruder to reach 100c
	G28 		; home all axes
	M561 		; clear any bed transform
	G1 Z5 F3000		;raise nozzle 5mm
	G1 X100 Y125 F10000   	;move to center of bed
	G30		;single probe and measure to get reference datum
	G29  		; probe the bed and enable compensation
	set global.bedOffset = 0.102 - floor(abs(abs(move.compensation.meshDeviation.deviation) + abs(move.compensation.meshDeviation.mean))*1000)/2000	;calculate the bed offset
M290 S{-move.axes[2].babystep}	;zero out babysteps
G28 ; home all axes
G92 E0 ; zero extruder
M290 S{global.bedOffset}	;apply bed offset
if (move.axes[2].babystep) = global.bedOffset
	M290 S0.10	;currently I need to add 0.10 for good 1st layer, may need to be dynamic based on layer height chosen
;M104 S223 T0		;heat extruder to 100c
;M109 S223 T0   		;wait for extruder to reach 100c
G1 E25 F225 ; purge nozzle
G92 E0 ; zero extruder
G1 Y20 F9600 ; move forward to avoid binder clips
G1 X190 Z0.1 E1.0 F1200 ; slow wipe
```

If the start macro isn't clear, it makes sure that there is no baby step offset before homing the axis, then it checks to see how many prints I've done and every 5th print, I do a bed mesh check to see if I've changed the bed mesh from removing the flex plate. Then that mesh value is averaged and used as a babyset step offset that gets applied before any motion happens to ensure the nozzle is the correct height above the bed.
